src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java

```diff
@@ -378,6 +319,72 @@ private long getNamespacePid(long pid) throws AttachNotSupportedException, IOExc
         }
     }
 
+    private static final String FIELD = "field";
+    private static final String MASK  = "mask";
+
+    private static final String SIGNAL_MASK_PATTERN = "(?<" + FIELD + ">Sig\\p{Alpha}{3}):\\s+(?<" + MASK + ">\\p{XDigit}{16}).*";
+
+    private static final long SIGQUIT = 1L << 2;
+
+    private static boolean checkCatchesAndSendQuitTo(int pid, boolean throwIfNotReady) throws AttachNotSupportedException, IOException {
+        var quitIgn = false;
+        var quitBlk = false;
+        var quitCgt = false;
+
+        final var procPid = PROC.resolve(Integer.toString(pid));
+
+        final var p = Pattern.compile(SIGNAL_MASK_PATTERN);
+
+        var readBlk = false;
+        var readIgn = false;
+        var readCgt = false;
```

Comment:
Nit:
 - Minor suggestion to order the lines 338-430 same as lines 330-332 and place them after them
 - Never liked single-letter identifiers because they are not well searchable. I'd suggest to:
   rename `p` to  something like `pattern` or `ptrn` and `m` to something like `matcher` or `match`.